File: CS50/Problem Set 1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int d_change(void)
{
    float c;
    printf("O hai! ");
    do
    {
        printf("How much change is owed?\n");
        c = get_float();
    }
    while (c < 0);
    int cents = (int) round(c * 100);
    return cents;
}

int main(void)
{
   // int coins = min_coins(ask_change());
   int cents = d_change();
   int q = 25;
    int d = 10;
     int n = 5;
     int p = 1;
    int coins = 0;
    while (cents > 0)
    {
        if (cents >= q)
        {
            cents -= q;
        }
        else if (cents >= d)
        {
            cents -= d;
        }
        else if (cents >= n)
        {
            cents -= n;
        }
        else if (cents >= p)
        {
            cents -= p;
        }
        coins++;
    }
    printf("%i\n", coins);
}

File: CS50/Problem Set 6/greedy.py
import cs50

def main():
    while True:
        print("O hai! How much change is owed?")
        amt = cs50.get_float()
        if amt >= 0:
            break
    
    num_of_coins = 0
    cents = int(round(amt * 100))
    
    num_of_coins += cents // 25
    cents %= 25
    
    num_of_coins += cents // 10
    cents %= 10
    
    num_of_coins += cents // 5
    cents %= 5
    
    num_of_coins += cents
    
    print("{}".format(num_of_coins))
    
if __name__ == "__main__":
    main()